[WIP] #4466 Port streams to bootstrap

pablocubico opened the pull request:
Hi!

This is a work in progress, some thing should be discussed, as you can see, we are using container-fluid which makes D\* go all the way to the screen width. I have a 1920x1080 monitor, so you can see how huge it gets. 

I don't know what's the "official" position about this, should we limit the width to some reasonable size? 

Also, we have to port the header.

Searching for "blueprint" on controllers and views gives no result. Not a scientific measurement, but I think we are really close to finally drop blueprint.

![image](https://cloud.githubusercontent.com/assets/340766/4388667/c067d720-43ed-11e4-808c-89bdfc10b3f5.png)

pablocubico committed:
main_stream ported, no further styling applied

pablocubico committed:
removing a missing mixing, using row instead of row-fluid

pablocubico committed:
left bar styles adjusted

pablocubico committed:
welcome to diaspora styles ported

pablocubico committed:
comment button using bootstrap styles

pablocubico committed:
colliding style removed

pablocubico committed:
removing blueprint classes from new aspect modal

pablocubico committed:
rightbar invitations ported to bootstrap

pablocubico committed:
dropping blueprint option on publisher

dimaursu commented on app/assets/stylesheets/invitations.css.scss:
```diff
@@ -5,11 +5,19 @@
 }
 
 #invitationsModal {
+  #paste_link_wrapper {
```
Styling on #id's is not recommended.

pablocubico replied:
Not a best practice, but in this case, I think it does a good job preventing any style collisions or cascading mistakes...

Anyway, using camelCase and under_scores mixed on two lines of code is not recommended either... :(

dimaursu commented on app/assets/stylesheets/new_styles/_registration.scss:
```diff
@@ -49,3 +49,34 @@
+  background: orange;
+  &:hover {
+    #gs-skip-x {
+      @include opacity(0.4);
```
opacity is supported unprefixed these days, 
`opacity: 0.4;` would be better. See #5532

pablocubico replied:
Will remove all those prefixes then.

dimaursu commented on app/assets/stylesheets/sidebar.css.scss:
```diff
@@ -187,7 +184,9 @@
       #invite_code {
         width: 100%;
-        @include box-sizing(border-box);
+        -webkit-box-sizing: border-box;
```
`box-sizing` should go unprefixed

dimaursu commented on app/views/aspects/new.haml:
```diff
@@ -2,27 +2,27 @@
-    %p.checkbox_select
-      = aspect.check_box :contacts_visible, :checked => false, class: 'checkbox'
-      = aspect.label :contacts_visible, t('aspects.edit.make_aspect_list_visible')
+      %br
```
Can't some `margin-bottom` be used here instead?

pablocubico replied:
Sure! Lazy me.

dimaursu commented on db/schema.rb:
```diff
@@ -300,8 +300,8 @@
     t.text     "parent_author_signature"
-    t.datetime "created_at"
-    t.datetime "updated_at"
+    t.datetime "created_at",              null: false
```
I suppose `db/schema.rb` has nothing to do with this pull-request, you should remove it ;)

pablocubico replied:
Oops! :)

dimaursu commented on app/assets/stylesheets/invitations.css.scss:
```diff
@@ -5,11 +5,19 @@
 }
```
a little bit up here, in you should set the heigth of the `#invite_code` field
`height: 1.6rem;`
worked fine for me, it's large enough that you can see the `p` descendant in `http` from the invite.
It's cool to user rems, because if you increase the font-size on the `html` or `body` element, the height will grow proportionally. rems, ems and percentages are responsive.

pablocubico replied:
I'm actually against using relative font-sizes as percentages, ems, or rems. Is a nice feature but in my experience it was only useful when having to resize something dinamycally as old newspaper sites did with the text body. Ems/rems also make hard to tell what a font size will look like when reading/writing code.

Generally, I prefer using specific pixel size for the fonts, and change it (if needed) in other form factors/viewport sizes by media queries.

Raven24 replied:
my (possibly outdated) view on this issue is that a web-design should try to be responsive enough to fit relative size spezifications. there is no gain in aiming for pixel-perfect rendering for any specific output device as rendering will always have different "flavours" and you can never predict how changing display densities or zooming will behave.

dimaursu replied:
@pablocubico One way is to set `font-size: 62.5%` on <html> - this makes any font below to act somewhat more familiar: 1.2rem == 12px, 1.4rem == 14px, and so on. Anyway, I think after moving to BS3 we can change them, it's not a big problem.

pablocubico replied:
In my experience doing responsive design for a few years now, there is gain in seeking pixel perfection: you have greater control and it ends up looking better. In most cases, I work with 3 basic scenarios/form factors: smartphone, tablet, desktop, and adjust the sizes accordingly. Specially on smartphones, which are a totally different use case.

dimaursu commented:
The content creation form looks way better than before. Good job, @pablocubico ;-)

AugierLe42e commented:
@pablocubico : I really love the font you used on your mockup pics. I'd really like to see this as default design !

pablocubico commented:
Hey all! I hope we can merge this, I'll take a better look over the weekend.

I haven't been very cautious about best CSS practices since I think all the
CSS code needs a serious refactor and I took this as a simple *porting* of
the streams, not really an enhancement. If it ends up looking better, well,
that's even better!



On Thu, Jan 8, 2015 at 7:13 PM, Augier <notifications@github.com> wrote:

> @pablocubico <https://github.com/pablocubico> : I really love the font
> you used on your mockup pics. I'd really like to see this as default design
> !
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/diaspora/diaspora/pull/5247#issuecomment-69259408>.
>

jhass commented:
You still have [WIP] in the title, what are the things you still want to do here?

Polishing to be done from my perspective (not necessarily in this PR), tested on FF34:

* The left sidebar feels to wide, especially on the followed tags page:
![selection_003](https://cloud.githubusercontent.com/assets/141294/5673288/d9ccdf58-9799-11e4-8ea9-d1bd30901ce0.png)
  Maybe we should add some spacing like on the tag stream?
* Also visible in above screenshot, the invite code copy paste box looks weird.
* Weird little grey box next to Reshare
![selection_004](https://cloud.githubusercontent.com/assets/141294/5673305/1122c09e-979a-11e4-9f66-1acb2b8bf4b6.png)
* Ist that my eyes, my system or is the Comemnt button blurry? 
![selection_005](https://cloud.githubusercontent.com/assets/141294/5673312/34765c90-979a-11e4-87cc-6f3dbb48bd5c.png)
* The initial comment textarea is larger than before and expands now when you click into it.

pablocubico committed:
ensuring that the oembed element has the same size of the thumb

pablocubico committed:
text shadow made the buttons look blurry

pablocubico committed:
invite code box fix

pablocubico committed:
reshare border fix

pablocubico committed:
using container instead of container fluid for the stream, we can manage this
better when migrating to Bootstrap 3

pablocubico committed:
using entypo on the right section, better alignment and styles

pablocubico committed:
lame BR removed

pablocubico committed:
unprefixing opacity

pablocubico committed:
main stream widths with fluid container

pablocubico committed:
margin 0 was messing up the left bar

pablocubico committed:
border on the left of the stream and profile header

pablocubico committed:
using circled help icon

pablocubico commented:
Ok, I rebased and made some changes based on the feedback you gave me, namely:

@svbergerem 
- stream page is now  offset1+span2, span6, span2 !
- bold right section titles with entypo icons, better alignment
- stream container now has a left border, had to add the same to the profile header

@dimaursu 
- removed css prefixes
- removed that lame BR

@jhass 
- your eyes are fine, comment button had a white text-shadow and white text color, so it looked blurry
- video thing: it seems to happen with some videos, not all of them. It happens on joindiaspora also, I made a fix to make the iframe the same size as the thumb (it's on a separate commit)
- spinner: I haven't changed any code related to that, I'll take a look now with the rebase, but I can't get the spinner to show (need a proper testing DB with more data...) 
- comment textarea: weird, maybe a class name affected the behaviour. Can we check that on a different PR?


Forgot to remove the schema commit, doing it now.

jhass commented:
When  I write

> Polishing to be done from my perspective (not necessarily in this PR)
 
I mean that ;)

pablocubico commented:
Just removed the schema commit. 

@jhass Haha, I know, but those were really minor things, now it looks more solid.

jhass commented:
Some tests seem to need fixing:

* https://travis-ci.org/diaspora/diaspora/jobs/46636352#L564
* https://travis-ci.org/diaspora/diaspora/jobs/46636354#L400
* https://travis-ci.org/diaspora/diaspora/jobs/46636354#L411
* https://travis-ci.org/diaspora/diaspora/jobs/46636354#L673

Probably more after that but the last one crashed the cucumber suite.

jhass commented:
Oh, do the report post and ignore user icons still show up for you? For me only the Hide/Delete icon shows up.

Flaburgan commented:
The spinner doesn't display on any bootstrap stream: the tag page and the profile page are affected too.

pablocubico commented:
Thanks for the feedback, I'm checking those issues.

On Sun, Jan 11, 2015 at 5:33 PM, Fla <notifications@github.com> wrote:

> The spinner doesn't display on any bootstrap stream: the tag page and the
> profile page are affected too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/diaspora/diaspora/pull/5247#issuecomment-69510086>.
>

AugierLe42e commented:
@pablocubico : I'm creating my own pod. When al of this is finished, I *do* want this style on it :+1: 
![](https://cloud.githubusercontent.com/assets/340766/4838791/b5996276-5ff1-11e4-897d-6a53d8a5fe81.png)

AugierLe42e commented:
@pablocubico : if #2874 is still relevant, maybe you can solve it with this PR ?

jhass commented:
@AugierLe42e Can you elaborate a bit on why that *feature* would be related to this PR?

AugierLe42e commented:
Not fully related but I'm being pragmatic : as @pablocubico already ported the contact page to BS and that PR is the following of this work, then it should be done now or closed.
Plus, if it is not done while the UI is completely refactored, nobody will do it anymore.

svbergerem commented:
@AugierLe42e: @pablocubico did **not** port the contacts page to Bootstrap, that was @blase in #5118.

> if it is not done while the UI is completely refactored, nobody will do it anymore

The issue is still open and I came across that one a few days ago. I have some other things to do right now but if no one else will implement it, I'll do that someday.

AugierLe42e commented:
Ok, sry for the noise ;)

svbergerem commented:
@pablocubico I checked your code again. It still needs a rebase and you might have to fix some tests.

You changed the invitation modal which already had been ported to Bootstrap before. (you changed `app/assets/stylesheets/invitations.css.scss` and `app/views/invitations/new.html.haml`) Tbh. I think the old version looked better:
before:
![](https://cloud.githubusercontent.com/assets/3798614/6221508/b662ef16-b644-11e4-8280-878eb1968fd5.png)

after:
![](https://cloud.githubusercontent.com/assets/3798614/6221511/c1f95978-b644-11e4-8ba7-92899cb05099.png)

On the main stream the css for the invitation code in the right sidebar is also affecting the invitation code in the modal. The selector is not specific enough in this case. Here is how you could fix that:
```diff
diff --git a/app/assets/stylesheets/sidebar.css.scss b/app/assets/stylesheets/sidebar.css.scss
index d28b141..42af9ec 100644
--- a/app/assets/stylesheets/sidebar.css.scss
+++ b/app/assets/stylesheets/sidebar.css.scss
@@ -140,7 +140,7 @@
         }
       }
 
-      #invite_code {
+      & > #invite_code {
         width: 100%;
         box-sizing: border-box;
         font-size: 11px;
```
Also it would be great if you could squash some of the commits.

Apart from the mentioned points I think this PR should be ready to merge. We still need to do some polishing but I think that should happen in other PRs. There is quite a lot of stuff that depends on this PR so it would be great if we could merge it soon. Tell us if you need any help.

pablocubico commented:
Hi there, I'll be able to tackle this next weekend, I've been travelling
since FOSDEM, still on Berlin, but going home on wednesday.

Definitely will check Steffan's remarks and close it ASAP.

On Mon, Feb 16, 2015 at 9:52 PM, Steffen van Bergerem <
notifications@github.com> wrote:

> @pablocubico <https://github.com/pablocubico> I checked your code again.
> It still needs a rebase and you might have to fix some tests.
>
> You changed the invitation modal which already had been ported to
> Bootstrap before. (you changed app/assets/stylesheets/invitations.css.scss
> and app/views/invitations/new.html.haml) Tbh. I think the old version
> looked better:
> before:
>
> <https://cloud.githubusercontent.com/assets/3798614/6221508/b662ef16-b644-11e4-8280-878eb1968fd5.png>
>
> after:
>
> <https://cloud.githubusercontent.com/assets/3798614/6221511/c1f95978-b644-11e4-8ba7-92899cb05099.png>
>
> On the main stream the css for the invitation code in the right sidebar is
> also affecting the invitation code in the modal. The selector is not
> specific enough in this case. Here is how you could fix that:
>
> diff --git a/app/assets/stylesheets/sidebar.css.scss b/app/assets/stylesheets/sidebar.css.scss
> index d28b141..42af9ec 100644--- a/app/assets/stylesheets/sidebar.css.scss+++ b/app/assets/stylesheets/sidebar.css.scss@@ -140,7 +140,7 @@
>          }
>        }
> -      #invite_code {+      & > #invite_code {
>          width: 100%;
>          box-sizing: border-box;
>          font-size: 11px;
>
> Also it would be great if you could squash some of the commits.
>
> Apart from the mentioned points I think this PR should be ready to merge.
> We still need to do some polishing but I think that should happen in other
> PRs. There is quite a lot of stuff that depends on this PR so it would be
> great if we could merge it soon. Tell us if you need any help.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/diaspora/diaspora/pull/5247#issuecomment-74599899>.
>

AugierLe42e commented:
@pablocubico : do you want some help to finish this ?

pablocubico commented:
Feel free to help! I've been way too busy at work when I got back from
Europe, March will be hopefully a more quiet month.

On Wed, Feb 25, 2015 at 6:54 AM, Augier <notifications@github.com> wrote:

> @pablocubico <https://github.com/pablocubico> : do you want some help to
> finish this ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/diaspora/diaspora/pull/5247#issuecomment-75933347>.
>

AugierLe42e commented:
Just tell me what you need. I'll submit PR to your repo.

jhass commented:
@svbergerem polished up the remaining points in #5715. Thank you!
